projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/with-ai-assistant.tsx

```diff
@@ -54,18 +64,63 @@ export const withAIAssistant = createHigherOrderComponent(
 			[ clientId, updateBlockAttributes ]
 		);
 
-		useSuggestionsFromAI( { prompt: storedPrompt, onSuggestion: setContent } );
+		const addAssistantMessage = useCallback(
+			( assistantContent: string ) => {
+				setStoredPrompt( prevPrompt => {
+					/*
+					 * Add the assistant messages to the prompt.
+					 * - Preserve the first item of the array (`system` role )
+					 * - Keep the last 4 messages.
+					 */
+
+					// Pick the first item of the array.
+					const firstItem = prevPrompt.messages.shift();
+
+					const messages: Array< PromptItemProps > = [
+						firstItem, // first item (`system` by default)
+						...prevPrompt.messages.splice( -3 ), // last 3 items
```

Comment:
I know it's an arbitrary number to prevent the history from growing too much, just need to make sure the responses will not become strange due to "truncating" the history. It's a matter of test, test, test.